Lib/test/test_context.py

```diff
@@ -383,6 +383,31 @@ def sub(num):
             tp.shutdown()
         self.assertEqual(results, list(range(10)))
 
+    def test_token_contextmanager_with_default(self):
```

Comment:
Maybe more test with a re-entrant context? as well as a test when an exception is raised in the context body?

Reply:
Thanks for the review, I'll add tests.
I don't expect any problems though since the implementation is really trivial.

Reply:
Up to you though! Generally, I'm not actually those kind of tests to test the implementation but rather to spot possible regressions if we change something. I actually don't know if we are that pedantic when testing other context managers so feel free not to burden the tests if you think it's not worth it!

Reply:
Tests are cheap and easy to write,
I've added two: one for exit when an exception is raised and another for reentrancy.

Please feel free to ask for additional tests if needed.

Reply:
Looks good! Small question, but is there a necessity to check what happens if we call `c.reset()` inside the context manager? or multiple `c.set()` as well?

Reply:
Multiple `c.set()` doesn't affect the context manager, the value will be reset on exit.
`c.reset()` is safe if the other token was used; resetting with the same token raises RuntimeErorr as in the following example:
```
with c.set(1) as token:
    c.reset(token)
```

Reply:
A couple additional tests were added to demonstrate mentioned scenarios.